### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,20 +39,29 @@ Install it with:
...
 
-    set PATH="%AppData%\cabal\bin\;%PATH%"
-
+```batch
+: requires cabal-install 1.12, or later
+for /f %i in ('cabal path -v0 --installdir') do set PATH=%i;%PATH%
+```
+or
+```batch
+set PATH="%AppData%\cabal\bin\;%PATH%"
```
Comment (sol) [inline README.md:63]: @mpilgrem do you know if this is still the default location on windows? The Cabal docs only say `~/.local/bin`.
https://cabal.readthedocs.io/en/stable/config.html#directories

File: README.md [hunk 2]
```diff
@@ -39,20 +39,29 @@ Install it with:
...
+export PATH="$HOME/.local/bin:$PATH"
+```
 
 On Windows:
 
-    set PATH="%AppData%\cabal\bin\;%PATH%"
-
+```batch
+: requires cabal-install 1.12, or later
+for /f %i in ('cabal path -v0 --installdir') do set PATH=%i;%PATH%
```
Comment (sol) [inline README.md:59]: @mpilgrem The Internet™ says that this is apparently how to do things with `cmd.exe`.

1. Can you try if this works?
2. I kind of assume that this actually should be `for /f %i in ('cabal path -v0 --installdir') do set PATH="%i;%PATH%"` instead (note the quotes).  Does `%i` still get properly expended with the quotes?
3. Is the for-loop really required? Would something like `set PATH=('cabal -v0 --installdir');%PATH%` work instead?

File: README.md (hunk 1 above)
Comment (mpilgrem) [reply README.md:63]: The short answer is "Yes, but" - See `Distribution.Client.Config.defaultInstallPath` and `maybeGetCabalDir` of the `cabal-install` package; the latter uses `getAppUserDataDirectory "cabal"`.

The "but" is that I think the concept of 'default' depends on how the user manages versions of Cabal (the tool). If they use GHCup to do so (as I do), then `cabal path --installdir` yields `C:\cabal\bin`.

File: README.md (hunk 2 above)
Comment (mpilgrem) [reply README.md:59]: I am relying on [this documentation](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/for) about Command Prompt's `for`. It seems to me:
1.  the command whose output is being parsed needs to be between back-quotes and, I think, the use of the parsing key word `usebackq`;
2.  you need to be clear if this advice is at the command line or in a batch file: the former uses `%i`, the latter `%%i`;
3.  it is possible for a Windows path to contain spaces. I think that means you need the parsing key word `tokens=*` and the quotation marks around `%i;%PATH%`.

So, at the command line, I think it needs to be as follows:
~~~text
for /f "usebackq tokens=*" %i in (`cabal path -v0 --installdir`) do set PATH="%i;%PATH%"
~~~

but I have yet to test that.

Comment (mpilgrem) [reply README.md:59]: I was wrong about quotation marks around `%i;%PATH%`. I think this works at the command line
~~~text
for /f "usebackq tokens=*" %i in (`cabal path -v0 --installdir`) do set PATH=%i;%PATH%
~~~

File: README.md (hunk 1 above)
Comment (sol) [reply README.md:63]: Again, thanks a lot @mpilgrem.

> If they use GHCup to do so (as I do), then `cabal path --installdir` yields `C:\cabal\bin`.

OK, that makes sense. That's consistent with what I am seeing on GitHub actions.

I am going to only document the cabal 3.12 way for windows then. Admittedly that's not perfect currently, but as time goes by that will work for the majority of the users.

**Pull request merged**